build/build.go

```diff
@@ -600,6 +606,11 @@ func (db *DeploymentBuilder) buildPlatformNetworks(d *Deployment) error {
 
 	d.PlatformNetworks = nets
 
+	err = db.filterPlatformNetworks(nil, d)
```

Comment:
Why are we offering a nil value?

Reply:
We want to filter out core networks after having the complete list of deployment.PlatformNetworks.
Offering nil value signals CoreNetworkFilter.Filter can loop through once and for all for filtering out core networks.